src/Microsoft.DotNet.Build.CloudTestTasks/WriteBuildStatsJson.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    /// <summary>
+    /// Used to write build stats to a JSON file.
+    /// </summary>
+    public class WriteTestBuildStatsJson : Task
+    {
+        /// <summary>
+        /// Helper class to be serialized to JSON.
+        /// </summary>
+        class TestBuildStatsJson
+        {
+            internal class FailedProject
+            {
+                private string m_name;
+
+                public FailedProject(string name)
+                {
+                    m_name = name;
+                }
+
+                public string ProjectName { get { return m_name; } }
+            }
+
+            private List<string> m_aggregates;
+            private List<FailedProject> m_failedProjects;
```

Comment:
This violates the dotnet coding style guidelines (which I think apply to this project).